File: Easy/TwoSum.cs
//https://leetcode.com/problems/two-sum/solution/

public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        int[] answer = new int[2];
        
        var numberDictionary = new Dictionary<int,int>();
        
        for(int i = 0; i < nums.Count(); i++){
            var difference = target - nums[i];
            
            if(numberDictionary.ContainsKey(difference))
            {
                answer[0] = i;
                answer[1] = numberDictionary[difference];
            }
            else
            {
                numberDictionary.Add(nums[i], i);
            }
        }
        
        return answer;
    }
}

File: Easy/twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
                
        num_dict = {}
        
        for index, value in enumerate(nums):
            if value in num_dict:
                num_dict[value].append(index)
            else:
                num_dict[value] = [index]
                
        for index1, num1 in enumerate(nums):
            diff = target - num1
            if diff in num_dict:
                for index2 in num_dict[diff]:
                    if index2 != index1:
                        return [index1,index2]
